src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -300,11 +300,20 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
   }, [values]);
 
   useEffect(() => {
+    // When options are an array, then use them as-is.
+    // If options are an object, then use the `initial` array if this is the first time the field is opened.
+    // Otherwise, use the current fieldState array options.
+    const maybeUpdatedOptions = Array.isArray(maybeOptions)
+      ? maybeOptions
+      : firstOpen.current === false
+      ? fieldState.allOptions
+      : maybeOptions.initial;
+
     // Only update the fieldset when options change, when options is an array.
     // Otherwise, if the options are passed in as an object, then we assume the caller is updating options via a Promise and not via updating props.
```

Comment:
**request:** Can you either remove or update this comment. It is no longer relevant. Removing completely would probably be fine.

Reply:
Removed 👍